pulsar-common/src/main/java/org/apache/pulsar/common/util/FutureUtil.java

```diff
@@ -347,4 +347,21 @@ public static void safeRunAsync(Runnable runnable,
                     return null;
                 });
     }
+
+    public static <T> @Nonnull CompletableFuture<T> apply(@Nonnull CompletableFuture<T> s,
```

Comment:
please add a proper name and javadoc for this function. `completeAfter` would be a better name than `apply`.
a good guideline in naming is to be rather specific than generic. `apply` is very vague.

Reply:
you are right(naming is difficult...)